packages/hardhat-core/test/internal/hardhat-network/provider/HardhatBlockchain.ts

```diff
@@ -54,6 +61,17 @@ describe("HardhatBlockchain", () => {
       assert.equal(await blockchain.getBlock(one.hash()), one);
     });
 
+    it("can get existing block by hash (callbackified)", async () => {
+      const genesis = createBlock(0);
+      const one = createBlock(1);
+      await promisify<Block>(callbackifiedBlockchain.putBlock)(genesis);
+      await promisify<Block>(callbackifiedBlockchain.putBlock)(one);
```

Comment:
This is ugly, I'd rather have a `promisifiedCallbackifiedBlockchain` (without that name, I swear) and just duplicate the previous test as is.